optimizely/config_manager.py

```diff
@@ -30,282 +30,282 @@
-        self._set_config(datafile)
-
-    def _set_config(self, datafile):
-        """ Looks up and sets datafile and config based on response body.
-
-         Args:
-           datafile: JSON string representing the Optimizely project.
-         """
-
-        if self.validate_schema:
-            if not validator.is_datafile_valid(datafile):
-                self.logger.error(enums.Errors.INVALID_INPUT.format('datafile'))
-                return
-
-        error_msg = None
-        error_to_handle = None
-        config = None
-
-        try:
-            config = project_config.ProjectConfig(datafile, self.logger, self.error_handler)
-        except optimizely_exceptions.UnsupportedDatafileVersionException as error:
-            error_msg = error.args[0]
-            error_to_handle = error
-        except:
-            error_msg = enums.Errors.INVALID_INPUT.format('datafile')
-            error_to_handle = optimizely_exceptions.InvalidInputException(error_msg)
-        finally:
-            if error_msg:
-                self.logger.error(error_msg)
-                self.error_handler.handle_error(error_to_handle)
-                return
-
-        previous_revision = self._config.get_revision() if self._config else None
-
-        if previous_revision == config.get_revision():
+  """ Config manager that returns ProjectConfig based on provided datafile. """
+
+  def __init__(self,
+               datafile=None,
+               logger=None,
+               error_handler=None,
+               notification_center=None,
+               skip_json_validation=False):
+    """ Initialize config manager. Datafile has to be provided to use.
+
+    Args:
+        datafile: JSON string representing the Optimizely project.
+        logger: Provides a logger instance.
+        error_handler: Provides a handle_error method to handle exceptions.
+        notification_center: Notification center to generate config update notification.
+        skip_json_validation: Optional boolean param which allows skipping JSON schema
+                              validation upon object invocation. By default
+                              JSON schema validation will be performed.
+    """
+    super(StaticConfigManager, self).__init__(logger=logger,
+                                              error_handler=error_handler,
+                                              notification_center=notification_center)
+    self._config = None
+    self.validate_schema = not skip_json_validation
+    self._set_config(datafile)
+
+  def _set_config(self, datafile):
+    """ Looks up and sets datafile and config based on response body.
+
+     Args:
+       datafile: JSON string representing the Optimizely project.
+     """
+
+    if self.validate_schema:
+        if not validator.is_datafile_valid(datafile):
+            self.logger.error(enums.Errors.INVALID_INPUT.format('datafile'))
             return
 
-        self._config = config
```

Comment:
How are config manager changes related to event processor changes? Please remove these changes from this PR.